### DISCUSSION TITLE
Is there a way to call Tauri's API from a page embedded in an iframe?

### DISCUSSION THREAD

**Discussion by muwoo** [tauri-apps/tauri#9967] [Q&A] [answered]
It seems that the iframe embedded in tauri will not inject some global functions into the window, such as `window.__TAURI_IPC__`. This makes the `invoke` function unusable.

Is there any other way to let the page embedded in the iframe use Tauri's API?

**Comment by muwoo** [comment, marked as answer]
The issue has been resolved. Just recording my own solution for future reference: 

If your `iframe` and `main window` have the same domain name, then this may be a good method(Add the following code to the very top of `iframe` js initialization):

```js
// Override the __TAURI_IPC__ function
window.__TAURI_IPC__ = (args) => {
  //The parent window callback points to the child window
  window.parent[`_${args.callback}`] = window[`_${args.callback}`];

  // Call the function of the parent window 
  window.parent.__TAURI_IPC__(args);
};
```
First, `Tauri` will mount the global `window.__TAURI_IPC__` in the `main window` by default, but there is no such function in the child `iframe`. However, many of Tauri's APIs are based on the `window.__TAURI_IPC__` function to communicate with the Tauri backend, so we need to set a `window.__TAURI_IPC__` function for the child iframe.

However, the `callback` function of the communication is defined in the child iframe, so when the communication ends, the `callback` function cannot be found in the `parent window`, so we need to assign the `callback` function of the child iframe to the window object of the parent window.

**Comment by val-o** [comment]
Is there any similar way to do it in v2?
